### PR TITLE
Apply latest discussions

### PR DESCRIPTION
**Pull request by jihoonsong**
I tried to incorporate the latest discussions as much as I can. While I was writing, I came up with some thoughts:

- We can consider adding timeline of execution layer, too.
- Removed "Main FOCIL Functions" as "Valid" is now what's left. If we want, we can add a subsection that describing each steps in detail e.g., IL construction, IL broadcast prohibiting IL equivocation, block validation, etc.
- When we had inclusion summary in mind, there was a mechanism handling AA issue. I think this is addressed by checking if any missing transactions can be appended at the bottom of a block. But if I missed anything, AA handling mechanism might deserve its subsection under "Backwards Compatibility".

### CODE REVIEW

File: EIPS/eip-FOCIL.md [hunk 1]
```diff
@@ -26,34 +27,32 @@ FOCIL is a simple committee-based design improving upon previous IL mechanisms o
...
+### Execution Layer
+
+TBD
 
-#### High-level overview
+### Consensus Layer
 
 #### Timeline
 
-A set of validators is selected from the beacon committee to become IL committee members for `slot N`.
```
Comment (soispoke) [inline EIPS/eip-FOCIL.md:35]: - A set of validators is selected from the beacon committee to become IL committee members for `slot N`. 

remove: Their local inclusion lists constraint the block proposer for `slot N+1` (we will get to it later on)

File: EIPS/eip-FOCIL.md [hunk 2]
```diff
@@ -26,34 +27,32 @@ FOCIL is a simple committee-based design improving upon previous IL mechanisms o
...
-- **`Slot N+1`, `t=0`**: The block producer of `slot N+1` releases its `block B`, which contains both the payload and the IL aggregate.
-- **`Slot N+1`, `t=4`**: The attesters of `slot N+1` vote on `block B`. `Block B` is considered valid if:
-  - The IL aggregate included in `block B` corresponds to the attesters' local view of available local ILs and is the same IL aggregate broadcast during `slot N`
-  - The IL aggregate entries are satisfied by payload transactions
+- **`Slot N`, `t=0 to 9s`**: After processing the block for `slot N` and confirming it as the head, each IL committee member of `Slot N` constructs a local inclusion list based on the head and their view of the public mempool, then broadcasts it over the P2P network. If no block arrives by `t=8s`, each IL committee member should run `get_head` to retrieve their canonical head and use it to build their local inclusion list.
```
Comment (soispoke) [inline EIPS/eip-FOCIL.md]: remove: If no block arrives by `t=8s`, each IL committee member should run `get_head` to retrieve their canonical head and use it to build their local inclusion list.

File: EIPS/eip-FOCIL.md [hunk 3]
```diff
@@ -11,12 +11,13 @@ created: 2024-07-??
 ---
 
 ## Abstract
```
Comment (soispoke) [inline EIPS/eip-FOCIL.md:13]: ## Abstract
Implement a robust mechanism to preserve Ethereum’s censorship resistance and chain neutrality properties by guaranteeing timely transaction inclusion. 

FOCIL is built in a few simple steps:
- In each slot, a set of validators is selected to become IL committee members. Each member gossips one local inclusion list according to their subjective view of the mempool.
- The proposer and all attesters of the next slot monitor, forward and collect available local inclusion lists. 
- The proposer includes transactions from all collected local ILs in its block before broadcasting it to the rest of the network.
- Attesters only vote for the proposer's block if it includes transactions from local inclusion lists they collected.

File: EIPS/eip-FOCIL.md (hunk 2 above)
Comment (soispoke) [inline EIPS/eip-FOCIL.md]: change from `t=0 to 9s` to `t=0 to 8s`

File: EIPS/eip-FOCIL.md [hunk 4]
```diff
@@ -26,34 +27,32 @@ FOCIL is a simple committee-based design improving upon previous IL mechanisms o
...
 
-### Consensus layer
+### Execution Layer
+
+TBD
 
-#### High-level overview
+### Consensus Layer
 
 #### Timeline
```
Comment (soispoke) [inline EIPS/eip-FOCIL.md:38]: Could you rewrite that part following the "participant/role" structure in the https://ethresear.ch/t/focil-cl-el-workflow/20526 doc?

File: EIPS/eip-FOCIL.md [hunk 5]
```diff
@@ -63,13 +62,21 @@ This EIP introduces backward incompatible changes to the block validation rule s
...
+It is important to ensure there is enough time between the local inclusion list freeze deadline (`t=9s` of `slot N`) and the moment at which the block producer has to broadcast `block B`, so that there is enough time to update `block B`'s execution payload according to the observed IL constraints.
+
+### IL Equivocation
+
+Since the local inclusion lists from the IL committee are all different and FOCIL does not introduce any single actor with sole responsibility, it seems infeasible to aggregate local inclusion lists while satisfying the aforementioned core properties. A malicious IL committee member may equivocate their local inclusion list.
+
+To mitigate local inclusion list equivocation, FOCIL introduces a new P2P network rule that allows forwarding up to two local inclusion lists per IL committee member. If the block proposer or attesters detect two different local inclusion lists sent by the same IL committee member, they should ignore all local inclusion lists from that member. In the worst case, the bandwidth of the local inclusion list gossip subnet can at most double.
+
+### IL Stuffing
```
Comment (soispoke) [inline EIPS/eip-FOCIL.md]: Could your remove this part? 

and instead add a small paragraph about invalidation?

**Review by soispoke — COMMENTED** [review]
Most important update would be to rewrite the timeline by including the "role/participant" structure in the https://ethresear.ch/t/focil-cl-el-workflow/20526 post

File: EIPS/eip-FOCIL.md (hunk 3 above)
Comment (jihoonsong) [reply EIPS/eip-FOCIL.md:13]: For me, "include transactions from all local ILs" sounds a bit flaky. It says no condition. It becomes more open to interpretation than before.

File: EIPS/eip-FOCIL.md (hunk 2 above)
Comment (jihoonsong) [reply EIPS/eip-FOCIL.md]: 9s is correct as IL committee can broadcast their local IL between 8s and 9s, no?

File: EIPS/eip-FOCIL.md (hunk 3 above)
Comment (soispoke) [reply EIPS/eip-FOCIL.md:13]: it's the abstract, the idea is to give an intuition here that people can quickly understand, before diving into the details later

File: EIPS/eip-FOCIL.md (hunk 2 above)
Comment (soispoke) [reply EIPS/eip-FOCIL.md]: Here, the idea is to have IL committee members stop sending local ILs 1 sec (t=8sec) before the freeze deadline so we make local ILs have time to propagate before the freeze deadline (at t=9sec). 

this should be changed accordingly too: If no block arrives by `t=7s`, each IL committee member should run `get_head` to retrieve their canonical head and use it to build their local inclusion list.

Comment (jihoonsong) [reply EIPS/eip-FOCIL.md]: I know the idea but isn't the IL deadline still 9s? The description between 0-9s follows IL committee members section in the [post](https://ethresear.ch/t/focil-cl-el-workflow/20526).

Comment (soispoke) [reply EIPS/eip-FOCIL.md]: Yeah yeah I know it's different than the post, but after discussions we think it's better to have a buffer so local Ils have time to propagate before the freeze deadline (1 sec should be enough)

Comment (jihoonsong) [reply EIPS/eip-FOCIL.md]: Ah, so there was some change that I haven't heard of. Applied it, thanks. Just give me heads up next time :)

**Review by soispoke — APPROVED** [review]

**Pull request merged**